`tests/unit.rs` broken on nightly builds

Travis: https://travis-ci.org/servo/rust-url/jobs/248944414

Example error:

```
error[E0560]: struct `__test::test::TestDesc` has no field named `allow_fail`
  --> tests/unit.rs:20:1
   |
20 | / fn size() {
21 | |     use std::mem::size_of;
22 | |     assert_eq!(size_of::<Url>(), size_of::<Option<Url>>());
23 | | }
   | |_^ `__test::test::TestDesc` does not have this field
```

I think it's caused by `tests/unit.rs` having test harness enabled, and the test internals having changed and not conflicting `rustc-test` package used here.

I'm not sure what's the best option for a fix:
* Disabling harness for `unit.rs` and copying in boilerplate to get the tests running; or
* Updating `rustc-test` to make it compatible.

What do you think, @SimonSapin ?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,7 +31,7 @@ harness = false
 test = false
 
 [dev-dependencies]
-rustc-test = "0.1"
+rustc-test = "0.2"
 rustc-serialize = "0.3"
 serde_json = ">=0.6.1, <0.9"
 

--- a/idna/Cargo.toml
+++ b/idna/Cargo.toml
@@ -18,7 +18,7 @@ harness = false
 name = "unit"
 
 [dev-dependencies]
-rustc-test = "0.1"
+rustc-test = "0.2"
 rustc-serialize = "0.3"
 
 [dependencies]

--- a/idna/tests/tests.rs
+++ b/idna/tests/tests.rs
@@ -10,11 +10,7 @@ fn main() {
     {
         let mut add_test = |name, run| {
             tests.push(test::TestDescAndFn {
-                desc: test::TestDesc {
-                    name: test::DynTestName(name),
-                    ignore: false,
-                    should_panic: test::ShouldPanic::No,
-                },
+                desc: test::TestDesc::new(test::DynTestName(name)),
                 testfn: run,
             })
         };

--- a/tests/data.rs
+++ b/tests/data.rs
@@ -188,11 +188,7 @@ fn main() {
     {
         let mut add_one = |name: String, run: test::TestFn| {
             tests.push(test::TestDescAndFn {
-                desc: test::TestDesc {
-                    name: test::DynTestName(name),
-                    ignore: false,
-                    should_panic: test::ShouldPanic::No,
-                },
+                desc: test::TestDesc::new(test::DynTestName(name)),
                 testfn: run,
             })
         };

